### ISSUE TITLE
Possible Memory Leak

### ISSUE THREAD

**Issue by Cpruce** [awslabs/keras-apache-mxnet#195]
Please make sure that the boxes below are checked before you submit your issue.
If your issue is an **implementation question**, please ask your question on [StackOverflow](http://stackoverflow.com/questions/tagged/keras) or [on the Keras Slack channel](https://keras-slack-autojoin.herokuapp.com/) instead of opening a GitHub issue.

Thank you!

- [ X] Check that you are up-to-date with the master branch of Keras. You can update with:
`pip install git+git://github.com/keras-team/keras.git --upgrade --no-deps`

- [X ] Check that your version of TensorFlow is up-to-date. The installation instructions can be found [here](https://www.tensorflow.org/get_started/os_setup).

- [X ] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

Please see:

https://discuss.mxnet.io/t/possible-memory-leak/1973

**Comment by Cpruce** [comment]
@roywei Thanks for looking into this

**Comment by roywei** [comment]
@Cpruce I was able to narrow down the memory leak at validation time after each epoch. For now, removing validation during model.fit() resolved this, and use model.evaludate(test_data, test_label) to do validation at the end works fine. 
We are using bucketing module in keras-mxnet, maybe switching bucket between train and validation caused the memory leak in foreach operator. Need to take another look at that.

**Comment by Cpruce** [comment]
@roywei awesome thanks I'll try it out soon 👍

**Comment by roywei** [comment]
For now removing validation dataset resolves the memory leak issue
using the following command for training:
```
history = model1.fit(x_train, y_train,
                    epochs=epochs,
                    batch_size=batch_size,
                    callbacks=[reduce_lr],
                    verbose=2)
```
need to investigate on how to re-enbale validation stage

**Comment by julioasotodv** [comment]
I can confirm that the memory leak is happening in mxnet-mkl 1.13.1 under Linux, when running the imdb_bidirectional_lstm.py in the examples folder (which includes a validation set)
